src/btree/node.rs

```diff
@@ -0,0 +1,289 @@
+// This file implements our B+Tree nodes, which wrap flatbuffers.
+
+// TODO remove this once this code is called from repc.
+#![allow(dead_code)]
+
+use super::nodes_generated::nodes as fb;
+use std::cmp::Ordering;
+use std::iter::Iterator;
+use str_macro::str;
+
+// TODO:
+//   - read/write to/from Chunk
+//   - validate node on load
+//   - implement insert() into a Node
+//   - propagate hash and/or max key change from a node to its parents
+//   - save to/from the dag
+//   - provide degenerate 2-level btree::DagMap and replace prolly::Map with it
+//   - provide a nice way to print a Node and DagMap
+//   - turn btree::DagMap into a full B+Tree
+
+// Node is a node in our B+Tree. A Node wraps a flatbuffer node of the corresponding
+// type. A Leaf node is a special case of InternalNode.
+#[derive(Debug)]
+pub enum Node<'a> {
+    Internal(fb::InternalNode<'a>),
+    Leaf(fb::InternalNode<'a>),
+    Data(fb::DataNode<'a>),
+}
+
+impl Node<'_> {
+    // key_iter returns an Iterator that iterates the keys contained in the node.
+    pub fn key_iter<'b, 'a: 'b>(&'a self) -> Box<dyn Iterator<Item = &'a [u8]> + 'b> {
```

Comment:
Someone is going to suggest why not just have an iter over both key and value and ignore the value when it's not needed. Good suggestion! I think when I started this I mistakenly thought for some reason I shouldn't iterate the values, like in case they are large. But since the value is just a reference to something it's not going to be large itself. I'll make that change in the next PR.

Reply:
Oh, I remember now. The values in data nodes are [u8] but the values internal nodes are chunk hashes aka &strs, so not obvious how to share an interator.